mesonbuild/minstall.py

```diff
@@ -217,6 +224,22 @@ def check_for_stampfile(fname):
                 return files[0]
     return fname
 
+def setup_dry_run():
+    global dry_run
+    dry_run = True
+    noop = lambda *args, **kwargs: None
+    os.chmod = noop
+    os.chown = noop
+    os.makedirs = noop
+    os.mkdir = noop
+    os.remove = noop
+    os.symlink = noop
+    shutil.copy = noop
+    shutil.copyfile = noop
+    shutil.copystat = noop
+    depfixer.fix_rpath = noop
```

Comment:
I don't like monkey patching the running process like this. One thing I have wanted to do for a while is to run some, preferably all, Meson tool invocations in unittests as direct function calls rather than spawning processes. If you do that, these changes persist (at least I think they do) and can cause interesting things in the caller later on.

Reply:
Yes, that's probably exactly why the tests fail right now. I think we should use a wrapper for all these functions and do nothing if dry_run is set. Monkey-patching is genius, but also brittle.

Reply:
Ok, I agree. I wanted to make sure we can't sneak an os.copy somewhere without noticing, but it's cleaner to keep a wrapper indeed. Patch updated.